Here is the promised patch to zip 1.0. The source for zip is available
at wsmr-simtel20.army.mil: pd1:<misc.unix>zip10ex.tar-z. The next
release of zip will have linux support included.

You need at least gcc2.12 for this to work. If not, zip dies with an
internal logic error.

-- 
Humberto Ortiz-Zuazaga                            zuazaga@ucunix.san.uc.edu
Dept. of Physiology & Biophysics                   University of Cincinnati

*** makefile.orig	Thu May 28 10:37:05 1992
--- makefile	Thu May 28 10:51:07 1992
***************
*** 7,13 ****
  	@echo ''
  	@echo 'Make what?  You must say what system to make Zip for--e.g.'
  	@echo '"make bsd".  Choices: bsd, bsdold, sysv, sun, next, next10,'
! 	@echo 'hpux, dnix, cray, 3b1, zilog, aux, convex, aix, and minix.'
  	@echo 'See the file zip.doc for more information.'
  	@echo ''
  
--- 7,13 ----
  	@echo ''
  	@echo 'Make what?  You must say what system to make Zip for--e.g.'
  	@echo '"make bsd".  Choices: bsd, bsdold, sysv, sun, next, next10,'
! 	@echo 'hpux, dnix, cray, 3b1, zilog, aux, convex, aix, linux and minix.'
  	@echo 'See the file zip.doc for more information.'
  	@echo ''
  
***************
*** 137,142 ****
--- 137,146 ----
  aix:
  	$(MAKE) zips CC="c89" BIND="c89" \
  	   CFLAGS="-O -D_POSIX_SOURCE -D_ALL_SOURCE -D_BSD -DAIX"
+ 
+ # Linux 0.96a and GCC 2.12b.
+ linux:
+ 	$(MAKE) zips CFLAGS="-O2 -DLINUX" CC=gcc BIND=gcc
  
  # MINIX 1.5.10 with Bruce Evans 386 patches and gcc/GNU make
  minix:
*** fileio.c.orig	Thu May 28 10:36:47 1992
--- fileio.c	Thu May 28 10:57:09 1992
***************
*** 185,191 ****
       DIR *opendir OF((char *));
  #  endif /* !NODIR */
  #  ifndef CONVEX
!      struct direct *readdir OF((DIR *));
  #  endif /* !CONVEX */
  #endif /* ?DIRENT */
  #endif /* !UTIL */
--- 185,193 ----
       DIR *opendir OF((char *));
  #  endif /* !NODIR */
  #  ifndef CONVEX
! #    ifndef LINUX
!        struct direct *readdir OF((DIR *));
! #    endif /* !LINUX */
  #  endif /* !CONVEX */
  #endif /* ?DIRENT */
  #endif /* !UTIL */
***************
*** 197,203 ****
  int unlink OF((char *));
  #ifndef CONVEX
  #  ifndef AIX
!      int chmod OF((char *, int));
  #  endif /* !AIX */
  #endif /* !CONVEX */
  
--- 199,207 ----
  int unlink OF((char *));
  #ifndef CONVEX
  #  ifndef AIX
! #    ifndef LINUX
!        int chmod OF((char *, int));
! #    endif /* !LINUX */
  #  endif /* !AIX */
  #endif /* !CONVEX */
  
